internals/integrations/tfstate/backend_other.go

```diff
@@ -0,0 +1,185 @@
+// +build !windows !386
+
+package tfstate
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secretpath"
+)
+
+type backend struct {
+	client secrethub.ClientInterface
+	port   uint16
+	logger io.Writer
+}
+
+func New(client secrethub.ClientInterface, port uint16, logger io.Writer) Backend {
+	return &backend{
+		client: client,
+		port:   port,
+		logger: prefixWriter{
+			Writer: logger,
+			prefix: "[SecretHub]: ",
+		},
+	}
+}
+
+func (b *backend) Serve() error {
+	server := &http.Server{
+		Addr:    fmt.Sprintf("127.0.0.1:%d", b.port),
+		Handler: http.HandlerFunc(b.Handle),
+	}
+	err := server.ListenAndServe()
+	if err != nil && err != http.ErrServerClosed {
+		return err
+	}
+
+	return nil
+}
+
+func (b *backend) Handle(w http.ResponseWriter, r *http.Request) {
+	resp, err := b.handle(r)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(b.logger, "Encountered an unexpected error: %s\n", err)
+		return
+	}
+	w.WriteHeader(resp.code)
+	fmt.Fprintf(w, resp.body)
+}
+
+type statusResponse struct {
+	code int
+	body string
+}
+
+func (b *backend) respondError(statusCode int, format string, a ...interface{}) *statusResponse {
+	msg := fmt.Sprintf(format, a...)
+	fmt.Fprintf(b.logger, "%s\n", msg)
+	return &statusResponse{
+		code: statusCode,
+		body: msg,
+	}
+}
+
+func (b *backend) handle(r *http.Request) (*statusResponse, error) {
+	isChild, err := connectionFromChildProcess(os.Getpid(), r)
+	if err != nil {
+		return nil, err
+	}
+	if !isChild {
+		return b.respondError(http.StatusForbidden, "can only be reached from a process spawned with secrethub run"), nil
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading request body: %s", err)
+	}
+
+	path, password, ok := r.BasicAuth()
+	if !ok {
+		return b.respondError(http.StatusBadRequest, "set the SecretHub path to the state as the username"), nil
+	}
+
+	if secretpath.Count(path) < 2 {
+		return b.respondError(http.StatusBadRequest, "set user to a valid repository or directory, got: %s", path), nil
+	}
+
+	statePath := secretpath.Join(path, "state")
+	lockPath := secretpath.Join(path, "lock")
+	passwordPath := secretpath.Join(path, "password")
+
+	secret, err := b.client.Secrets().ReadString(passwordPath)
+	if err != nil && !api.IsErrNotFound(err) {
+		return nil, err
+	} else if err == nil {
+		if password == "" {
+			return b.respondError(http.StatusUnauthorized, "password stored at %s should be set as auth password", passwordPath), nil
+		}
+		if password != secret {
+			return b.respondError(http.StatusForbidden, "provided password does not password stored at %s", passwordPath), nil
```

Comment:
`provided password does not password stored at %s` => `provided password does not match the password stored at %s`

Reply:
Good catch :ok_hand: